lib/pure/typetraits.nim

```diff
@@ -15,106 +15,152 @@
-  ## components or the number of generic parameters a given type ``t`` has.
+  ## components or the number of generic parameters a given type `t` has.
   runnableExamples:
-    assert arity(seq[string]) == 1
-    assert arity(array[3, int]) == 2
-    assert arity((int, int, float, string)) == 4
+    doAssert arity(int) == 0
+    doAssert arity(seq[string]) == 1
+    doAssert arity(array[3, int]) == 2
+    doAssert arity((int, int, float, string)) == 4
 
 proc genericHead*(t: typedesc): typedesc {.magic: "TypeTrait".} =
   ## Accepts an instantiated generic type and returns its
   ## uninstantiated form.
   ## A compile-time error will be produced if the supplied type
   ## is not generic.
   ##
-  ## See also:
-  ## * `stripGenericParams <#stripGenericParams,typedesc>`_
+  ## **See also:**
+  ## * `stripGenericParams proc <#stripGenericParams,typedesc>`_
   runnableExamples:
     type
       Foo[T] = object
       FooInst = Foo[int]
       Foo2 = genericHead(FooInst)
+
     doAssert Foo2 is Foo and Foo is Foo2
     doAssert genericHead(Foo[seq[string]]) is Foo
     doAssert not compiles(genericHead(int))
 
     type Generic = concept f
       type _ = genericHead(typeof(f))
+
     proc bar(a: Generic): typeof(a) = a
+
     doAssert bar(Foo[string].default) == Foo[string]()
     doAssert not compiles bar(string.default)
 
     when false: # these don't work yet
       doAssert genericHead(Foo[int])[float] is Foo[float]
       doAssert seq[int].genericHead is seq
 
-proc stripGenericParams*(t: typedesc): typedesc {.magic: "TypeTrait".}
+proc stripGenericParams*(t: typedesc): typedesc {.magic: "TypeTrait".} =
   ## This trait is similar to `genericHead <#genericHead,typedesc>`_, but
-  ## instead of producing error for non-generic types, it will just return
+  ## instead of producing an error for non-generic types, it will just return
   ## them unmodified.
+  runnableExamples:
+    type Foo[T] = object
+
+    doAssert stripGenericParams(Foo[string]) is Foo
+    doAssert stripGenericParams(int) is int
 
 proc supportsCopyMem*(t: typedesc): bool {.magic: "TypeTrait".}
-  ## This trait returns true if the type ``t`` is safe to use for
+  ## This trait returns true if the type `t` is safe to use for
   ## `copyMem`:idx:.
   ##
   ## Other languages name a type like these `blob`:idx:.
 
-proc isNamedTuple*(T: typedesc): bool {.magic: "TypeTrait".}
-  ## Return true for named tuples, false for any other type.
+proc isNamedTuple*(T: typedesc): bool {.magic: "TypeTrait".} =
+  ## Returns true for named tuples, false for any other type.
+  runnableExamples:
+    doAssert not isNamedTuple(int)
+    doAssert not isNamedTuple((string, int))
+    doAssert isNamedTuple(tuple[name: string, age: int])
+
+proc distinctBase*(T: typedesc): typedesc {.magic: "TypeTrait".} =
+  ## Returns the base type for distinct types. This works only
+  ## for distinct types and produces a compile time error otherwise.
```

Comment:
> for distinct types and produces a compile time error otherwise.

unrelated to your PR but IMO we need to revisit this so that `distinctBase(int) is int`, it's the most useful pattern

Reply:
=> https://github.com/nim-lang/Nim/pull/16891